iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PrivilegedUserCheck.java

```diff
@@ -96,7 +97,7 @@ private void handle(CheckContext ctx, DockerImage dockerImage) {
         ctx.reportIssue(dockerImage.from(), MESSAGE_OTHER_IMAGE);
       }
     } else {
-      String user = lastUser.get().user().value();
+      String user = ArgumentUtils.resolve(lastUser.get().arguments()).value();
```

Comment:
Seeing this for the second time: WDYT about an interface like `HasArguments` which describes the method `arguments()`. So we can extend the ArgumentUtils to handle these classes by calling the `arguments()`. This way you would get rid of calling it several places.